controller_connections.go

```diff
@@ -295,17 +298,23 @@ func (c *controller) listen() {
 		tmpLogger.WithError(err).Error("controller.Start() unable to start limited listener")
 		return
 	}
-
+	defer tmpLogger.Debug("controller.listen() stopping")
 	c.listener = l
 
+	go func() { // the listener doesn't play well with immediately stopping
```

Comment:
I saw you explained that in your PR comment but I still failed to understand why this can't be in the for loop, what problem does it have that this go routine approach doesn't have>

Reply:
I originally implemented it that way because there was a bug in the accept loop where the fatal error didn't have a "return" statement, which meant there was an infinite loop of errors if the listener was closed first. I fixed that but forgot to revert the other code.

Reply:
I understand better now, thank you.